flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/AggregationCodeGenerator.scala

```diff
@@ -179,13 +214,19 @@ class AggregationCodeGenerator(
                |      ${aggMapping(i)},
                |      (${accTypes(i)}) accs.getField($i));""".stripMargin
           } else {
+            val setDataView = if (accConfig.isDefined && accConfig.get.isUseState) {
```

Comment:
This will create new `MapState` (or `ListState`) objects in every invocation of `setAggregationResults()`. I think we can make the state objects members of the `GeneratedAggregations` class and just set them into the accumulator.